ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -164,69 +174,59 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
             if (streamSourceInfo.getStream() == null || streamSourceInfo.getStream().available() <= 0) {
                 throw new IngestionClientException("Stream is empty");
             }
-            String blobName = genBlobName("StreamUpload", ingestionProperties.getDatabaseName(), ingestionProperties.getTableName());
-            CloudBlockBlob blob = azureStorageHelper.uploadStreamToBlob(
+            String blobName = genBlobName(
+                    "StreamUpload", ingestionProperties.getDatabaseName(), ingestionProperties.getTableName());
+            CloudBlockBlob blob = azureStorageClient.uploadStreamToBlob(
                     streamSourceInfo.getStream(),
                     blobName,
                     resourceManager.getIngestionResource(ResourceManager.ResourceType.TEMP_STORAGE),
                     true
             );
-            String blobPath = azureStorageHelper.getBlobPathWithSas(blob);
-            BlobSourceInfo blobSourceInfo = new BlobSourceInfo(blobPath, 0); // TODO: check if we can get the rawDataSize locally
+            String blobPath = azureStorageClient.getBlobPathWithSas(blob);
+            BlobSourceInfo blobSourceInfo = new BlobSourceInfo(
+                    blobPath, 0); // TODO: check if we can get the rawDataSize locally
 
             ingestionResult = ingestFromBlob(blobSourceInfo, ingestionProperties);
             if (!streamSourceInfo.isLeaveOpen()) {
                 streamSourceInfo.getStream().close();
             }
             return ingestionResult;
+
         } catch (IOException | URISyntaxException e) {
-            throw new IngestionClientException("Error in ingestFromStream()", e);
+            throw new IngestionClientException("Failed to ingest from stream", e);
         } catch (StorageException e) {
-            throw new IngestionServiceException("Error in ingestFromStream()", e);
+            throw new IngestionServiceException("Failed to ingest from stream", e);
```

Comment:
Again, why service error?

Reply:
Also here, like above:
This is an Azure Storage exception. I think, for the user this is a Service exception, and not Client.